eventmesh-openconnect/eventmesh-openconnect-offsetmgmt-plugin/eventmesh-openconnect-offsetmgmt-api/src/main/java/org/apache/eventmesh/openconnect/offsetmgmt/api/storage/OffsetStorageWriterImpl.java

```diff
@@ -53,9 +53,12 @@ public OffsetStorageWriterImpl(String connectorName, OffsetManagementService off
     }
 
     @Override
-    public void writeOffset(RecordPartition partition, RecordOffset position) {
-        ConnectorRecordPartition extendRecordPartition = new ConnectorRecordPartition(connectorName, partition.getPartition());
-        data.put(extendRecordPartition, position);
+    public void writeOffset(RecordPartition partition, RecordOffset offset) {
+        ConnectorRecordPartition extendRecordPartition = null;
+        if (partition != null && partition.getPartition() != null) {
+            extendRecordPartition = new ConnectorRecordPartition(connectorName, partition.getPartition());
```

Comment:
No need to combine the `partition.getPartition() != null` condition, because the `partition` member of `RecordPartition` is a HashMap by default and it won't be null. 

If it is set to null mistakenly, then an exception should be thrown from where it is called.

https://github.com/apache/eventmesh/blob/5cc065f1a29d7827a5243d81c32722e0cc85ebc9/eventmesh-openconnect/eventmesh-openconnect-offsetmgmt-plugin/eventmesh-openconnect-offsetmgmt-api/src/main/java/org/apache/eventmesh/openconnect/offsetmgmt/api/data/RecordPartition.java#L24-L32